### DISCUSSION TITLE
Quick Gdiff from a commit view

### DISCUSSION THREAD

**Discussion by raags** [tpope/vim-fugitive#1661] [Q&A] [answered]
Is there a shortcut to get the Gdiff for a file from the arbitrary commit that I'm looking at. 

One use case is to pull changes from a previous commit for a file. For e.g. selectively reverting some change.

One way I could do that is:
1. Gblame on the section of code that I want to check
2. Press enter and open the relevant commit. Also, rebalme to get the previous commit so that `Gdiff` will work. 
3. Use y-<ctrl+g> to yank the commit id
4. Close buffer, navigate to the original file, and use `:Gdiffsplit <commit>`

Can these steps be shortened with a shortcut in the commit view itself, to get the diff against the current file? Actually, this will need to be the previous commit for `Gdiff` to work.

**Comment by tpope** [comment, marked as answer]
After you press enter to open the relevant commit, the cursor should be on the relevant `+` line in the diff for the file. If you move to a nearby `-` line and press enter again, it will take you to that line just before it was removed, which is in the previous commit as you desire. Call `:Gdiffsplit` from there to achieve your desired result.

*Edit:* If you don't want to seek out a nearby `-` line (because you're trying to script the whole process, for example), press enter on the `+` line, and then use `:Gedit >!^` to jump back one commit. The `!` means "the currently edited file's commit", `^` means "back one" as in Git, and `>` is a special notation to use the current filename and try to find the same line.

**Reply by raags** [reply]
Awesome! Is this documented, I couldn't find it the docs.

**Reply by tpope** [reply]
The individual pieces are, mostly (the explanation of `<CR>` is rather vague). A bit of a higher level explanation of `:Git blame` workflow would be nice, but I'm not sure where that should live. At least we have this publicly available discussion now.

**Reply by dannobytes** [reply]
Amazing 🤩

**Comment by idbrii** [comment]
`<CR>` opens a `:Gdiffsplit` when the cursor is on the 'diff --git' line of a commit view.

To open it from a blame:

1. Gblame on the section of code that I want to check
2. Press enter and open the relevant commit.
3. Use `?^diff<CR>` to search up to the start of this file's diff.
4. Hit `<CR>` to open a diff split for that file

You can make a mapping for it if you create ~/.vim/after/ftplugin/fugitiveblame.vim with:

    nmap <buffer> dd <CR>?^diff<CR><CR>

(We want CR remapped, so not using noremap.)
